[Data Grid] Add a recipe for handling long text cell (#20754)
mui/mui-x: docs/data/data-grid/cells/cells.md

Unlike `valueGetter()`, this function only impacts rendering—_not_ internal calculations like filtering or sorting.

See [Column definition—value formatter](/x/react-data-grid/column-definition/#value-formatter) for more details.

## Recipes

### Long text cell

> One approach to handling long text in a Data Grid cell is to create a custom cell that shows an expand button.

Reviewer: ```suggestion
To show long text in a Data Grid cell as a popup, you can create a custom cell that shows an expand button.
```
